src/orion/core/io/resolve_config.py

```diff
@@ -242,9 +249,43 @@ def merge_configs(*configs):
     return merged_config
 
 
-def infer_versioning_metadata(existing_metadata):
-    """Infer information about user's script versioning if available."""
-    # VCS system
-    # User repo's version
-    # User repo's HEAD commit hash
-    return existing_metadata
+def fetch_user_repo(user_script):
+    """Fetch the GIT repo and its root path given user's script."""
+    dir_path = os.path.dirname(os.path.abspath(user_script))
+    try:
+        git_repo = git.Repo(dir_path, search_parent_directories=True)
+    except git.exc.InvalidGitRepositoryError as e:
+        git_repo = None
+        raise Exception('Script should be a git repo') from e
```

Comment:
```python
('Script '{}' should be in a git repository.'.format(os.path.abspath(user_script)))
```

Reply:
Sorry, `RuntimeError` would be better actually. `SystemExit` should be at the *cli* level. See this [comment](https://github.com/mila-udem/orion/pull/115#issuecomment-414173033) for a bit more info.
```python
RuntimeError('Script '{}' should be in a git repository.'.format(os.path.abspath(user_script)))
```

The advantage of using `RuntimeError` is that a try-catch on it is more specific than `Exception`.